# U.S. Medical Insurance Costs

## Goals:
- Find the average cost between males and females --> subgrouping of smokers vs non-smokers
- Average age of someone with 1 child, 2 children, or 3+

In [23]:
# Initial import and storing data into a reiterable list
import csv

with open('insurance.csv', newline = '') as f:
    data_dict = csv.DictReader(f)
    data = []
    for row in data_dict:
        data.append(row)

Since we're interested in averages often, let's write a method to determine averages quickly for us!

In [24]:
def find_avg(target, list_of_data, target_key, target_val):
    total = 0
    count = 0
    
    for entry in list_of_data:
        if entry[target_key] == target:
            total += float(entry[target_val])
            count += 1
            
    return total / count

In [31]:
male_average = find_avg('male', data, 'sex', 'charges')
print(male_average)

female_average = find_avg('female', data, 'sex', 'charges')  
print(female_average)

13956.751177721886
12569.57884383534


|Sex | Average cost|
|---|---|
|Males | \$13,956.75|
|Females | \$12,569.58|

Next we'll find the average age of someone with no children, 1 child, 2 children, or 3 children

In [34]:
print(find_avg('0', data, 'children', 'age'))
print(find_avg('1', data, 'children', 'age'))
print(find_avg('2', data, 'children', 'age'))
print(find_avg('3', data, 'children', 'age'))

38.444250871080136
39.4537037037037
39.44583333333333
41.56687898089172


|Children | Average Age|
|---|---|
|0| 38.4|
|1| 39.5|
|2| 39.4|
|3+| 41.6|